rfcs/testdriver_bidi.md

```diff
@@ -0,0 +1,314 @@
+
+The specific protocol implementation would be required to opt-in for this protocol. Now it’s `EdgeChromiumDriverProtocol` and `ChromeDriverProtocol`.
+
+###### Dependency on `WebDriverProtocol`
+
+New [`WebDriverBidiProtocol`](https://github.com/web-platform-tests/wpt/pull/44649/files#diff-e5a8911dd97e0352b1b26d8ce6ef0a92b25378f7c9e79371a1eb1b1834bc9a8dR568) should be introduced. It is an extension of WebDriverProtocol, but with additional BiDi-specific protocol parts, e.g. [`BidiEventsProtocolPart`](https://github.com/web-platform-tests/wpt/pull/44649/files#diff-47192f72de48b834a50992ada793229686299b5165612c2d97af6811764514c7R325) and [`BidiScriptProtocolPart`](https://github.com/web-platform-tests/wpt/pull/44649/files#diff-47192f72de48b834a50992ada793229686299b5165612c2d97af6811764514c7R358). In order to enable the bidi session, the inherited `enable_bidi` flag should be overridden to `true`.
+
+###### Dedicated loop
+
+To be able to process events from the browser and from the tests in the same time, the WebDriverBidiProtocol protocol should have a dedicated [event loop](https://github.com/web-platform-tests/wpt/pull/44649/files#diff-e5a8911dd97e0352b1b26d8ce6ef0a92b25378f7c9e79371a1eb1b1834bc9a8dR577), which will be used by wptrunner to run async actions and process events in WebDriver BiDi transport.
+
+###### Alternative
+
+[Add `AsyncProtocolPart` with `get_loop` method](#add-asyncprotocolpart-with-get_loop-method).
+
+##### Extend `TestExecutor`
+
+In this example we consider changes in [`WebDriverTestharnessExecutor.do_testharness`](https://github.com/web-platform-tests/wpt/pull/44649/files#diff-e5a8911dd97e0352b1b26d8ce6ef0a92b25378f7c9e79371a1eb1b1834bc9a8dL591). Other `TestExecutor` (`MarionetteTestharnessExecutor`, `SeleniumTestharnessExecutor`, `ServoWebDriverTestharnessExecutor`,
+`WKTRTestharnessExecutor` etc) implementations should be done in the same way in order to support WebDriver BiDi.
+
+###### Messages from testdriver to wptrunner
+
+The communication channel between testdriver and wptrunner is set in the `do_testharness` method. Currently the messages from testdriver to wptrunner is done via long poll by calling `BaseProtocolPart.execute_async_script` which is synchronous (`async` in the name indicates that the script is wrapped in a special way before sent to the browser), and blocks processing the event loop in wptrunner, which in turn blocks the event processing while waiting for the next message form testdriver to wptrunner. To overcome this limitation, this RFC changes the wptrunner - testdriver communication channel to asynchronous non-blocking  [`BidiScriptProtocolPart.async_call_function`](https://github.com/web-platform-tests/wpt/pull/44649/files#diff-e5a8911dd97e0352b1b26d8ce6ef0a92b25378f7c9e79371a1eb1b1834bc9a8dR761) if available.  `WebDriverBidiProtocol`’s event loop should be used for message processing. This behavioral change is done only for protocols supporting BiDi.
```

Comment:
`async_call_function` seems like it could just be `call_function` since everything in bidi is async? That would also match the spec better.

Reply:
Renames